share/nisar/examples/noise_spectra_analysis.py

```diff
@@ -0,0 +1,448 @@
+"""
+Analyze noise-only receive power spectra. Collect spectral information with regards to
+RFI in terms of power and frequency locations.
+"""
+
+import numpy as np
+import argparse
+import h5py
+from scipy.fft import fftshift
+from scipy.signal import welch
+from numpy import linalg as la
+from nisar.products.readers.Raw import Raw
+from isce3.signal.compute_evd_cpi import slice_gen
+import os
+
+
+def cmd_line_parse():
+    parser = argparse.ArgumentParser(description='Perform spectra analysis on input L0B file.')
+
+    parser.add_argument(
+        '-i',
+        '--input', 
+        dest='input_file', 
+        type=str,
+        required=True, 
+        help='Input L0B file'
+    )
+    parser.add_argument(
+        '-o',
+        '--output', 
+        dest='output_dir', 
+        type=str,
+        required=True, 
+        help='Output data path. Output file name is input file name + SPECTRA.h5'
+    )
+    parser.add_argument(
+        "-p",
+        "--proc-interval",
+        dest="num_pulses_proc",
+        type=int,
+        required=False,
+        default="6",
+        help="Number of pulses to be used to estimate signle pulse power. Default: 6",
+    )
+    parser.add_argument(
+        "-n",
+        "--nfft",
+        dest="num_fft",
+        type=int,
+        default="2048",
+        help="Number of samples per segment for Scipy Welch function." 
+             "It is equal to number of FFT to be performed on each segment."
+             "Default: 2048"
+    )
+    parser.add_argument(
+        "-b",
+        "--block-size",
+        dest="num_pulses_blk",
+        type=int,
+        required=False,
+        default="4000",
+        help="Number of pulses per batch to be averaged in Azimuth-Time. Default: 5000",
```

Comment:
```suggestion
        default="4096",
        help="Number of pulses per batch to be averaged in Azimuth-Time. Default: 4096",
```
Powers of two will be an integer multiple of the HDF5 block size and therefore will allow more efficient reading. Since this size is much larger than the block size, it doesn't matter much, though.